prescient/simulator/config.py

```diff
@@ -74,334 +413,93 @@ def outer_parse(args=None, values=None):
-        domain=bool,
-        default=False,
-        description="Output sced initial conditions prior to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_sced_demands", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output sced demands prior to each solve. Default is False.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_solver_logs", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Output solver logs during execution.",
-    )).declare_as_argument()
-
-    simulator_config.declare("output_max_decimal_places", ConfigValue(
-        domain=PositiveInt,
-        default=6,
-        description="When writing summary files, this rounds the output to the "
-                    "specified accuracy. Default is 6.",
-    )).declare_as_argument()
-
-    simulator_config.declare("disable_stackgraphs", ConfigValue(
-        domain=bool,
-        default=False,
-        description="Disable stackgraph generation",
-    )).declare_as_argument()
-
-    return simulator_config
-
-PrescientConfig = create_Prescient_Config()
-
-def clear_prescient_config():
-    global PrescientConfig
-    PrescientConfig = create_Prescient_Config()
+class _PluginPath(Path):
+    ''' A Path that registers a plugin when its path is set
+    '''
+    def __init__(self, parent_config: PrescientConfig):
+        self.config = parent_config
+        super().__init__()
+
+    def __call__(self, data):
+        path = super().__call__(data)
+        self.config.plugin_context.register_plugin(path, self.config)
+        return path
+
+class _InEnumStr(InEnum):
+    ''' A bit more forgiving string to enum parser
+    '''
+    def __call__(self, value):
+        if isinstance(value, str):
+            value = value.replace('-', '_').upper()
+        return super().__call__(value)
+
+def _StartDate(data):
+    ''' A basic start date validator/converter
+    '''
+    if isinstance(data, date):
+        return data
+    elif isinstance(data, datetime):
+        if data.hour != 0 or data.minute != 0:
+            print(f"WARNING: Prescient simulations always begin a midnight; ignoring time {data.time()}")
+        return data.date()
+    try:
+        ans = dateutil.parser.parse(data).date()
+    except ValueError:
+        print(f"***ERROR: Illegally formatted start date={data} supplied!")
+        raise
+    return ans
+
+def _SolverOptions(data):
+    ''' A basic solver options validator.
+        Converts string options into a dictionary;
+        otherwise requires a dictionary.
+    '''
+    if isinstance(data, str):
+        ans = {}
+        opts = data.split(' ')
+        for opt in opts:
+            option, val = opt.split('=')
+            try:
+                val = float(val)
+            except:
+                pass
+            ans[option] = val
+        return ans
```

Comment:
This could be a problemmatic parser, as it will get fouled up by spaces in the arguments.  Maybe not an issue for solver options, although things like paths (i.e., to SOLN dumps or log files) are notorious for containing spaces (especially on Windows).  It might be overkill, but there are some interesting regex expressions that you can use (see, e.g. [here](https://stackoverflow.com/questions/27575708/how-to-split-comma-separated-key-value-pairs-with-quoted-commas/38787616) or [here](https://stackoverflow.com/questions/38737250/extracting-key-value-pairs-from-string-with-quotes)).  In Pyomo, we have broken down and just used `ply` (see [here](https://github.com/Pyomo/pyomo/blob/c91f821918fef7822e42c47bae73f4aa7c17387e/pyomo/core/base/componentuid.py#L609)).  Regardless, the approach is the same as yours: tokenize into key=value objects, then split on the first `=`.

This really begs for a utility method (in Pyomo?) that we can (re) use for parsing this kind of thing.

Reply:
I opted to use `shlex` for this task (a "battery" I didn't know about), which is easier (for me) to understand, while still providing more robustness than the original fragile implementation.